### PR TITLE
apply job

### PR DESCRIPTION
**Pull request by yongsokheng**

### CODE REVIEW

File: app/models/apply_job.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class ApplyJob < ApplicationRecord
+  belongs_to :user
+  belongs_to :job_post
+
+  PARAMS_APPLY_JOB = [:job_post_id]
+
+  validates_uniqueness_of :user_id, scope: :job_post_id
```
Comment (NgengChhengkim) [inline app/models/apply_job.rb:7]: co can validate khong?  belongs_to user, va job_post thi no se chi co 1 cap user, job_post duy nhat

Comment (yongsokheng) [reply app/models/apply_job.rb:7]: neu ko validate se luu duoc khong duy nhat

Comment (NgengChhengkim) [reply app/models/apply_job.rb:7]: thu chua?

Comment (yongsokheng) [reply app/models/apply_job.rb:7]: thu roi :smile:

File: app/models/apply_job.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  delegate :id, to: :job_post, prefix: true, allow_nil: true
+
+  scope :select_by_job, ->job_post_id{where job_post_id: job_post_id}
```
Comment (NgengChhengkim) [inline app/models/apply_job.rb:11]: `find_by job_post_id: job_post_id` hoac co the tim `ApplyJob.find_by user_id: user_id, job_post_id: job_post_id` de kiem tra

File: app/models/apply_job.rb (hunk 1 above)
Comment (NgengChhengkim) [reply app/models/apply_job.rb:7]: khong tin

File: app/views/apply_jobs/create.js.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<% if @apply_job.errors.any? %>
+  confirm_text = "<div class='alert alert-danger'><%= t 'errors.something_went_wrong' %></div>"
+<% else %>
+  confirm_text = "<div class='alert alert-success'><%= t 'new_feeds.labels.apply_success' %></div>"
+  $(".job-post-<%= @job_post_id %>")
+    .html("<%= j render 'shared/apply_job_action', job_post_id: @job_post_id %>");
+<% end %>
+
+$(".btn-apply").remove();
+$("#apply-confirm-modal").find(".content").html(confirm_text);
```
Comment (NgengChhengkim) [inline app/views/apply_jobs/create.js.erb]: $("#apply-confirm-modal .content")

File: db/migrate/20160927140142_create_apply_jobs.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateApplyJobs < ActiveRecord::Migration[5.0]
+  def change
+    create_table :apply_jobs do |t|
+      t.references :user, foreign_key: true
```
Comment (NgengChhengkim) [inline db/migrate/20160927140142_create_apply_jobs.rb]: khong add index?

Comment (NgengChhengkim) [reply db/migrate/20160927140142_create_apply_jobs.rb]: nhieu file khac nua toan khong co index

File: app/models/apply_job.rb (hunk 2 above)
Comment (yongsokhengkhmer) [reply app/models/apply_job.rb:11]: why? :smile:

File: app/models/apply_job.rb (hunk 1 above)
Comment (yongsokhengkhmer) [reply app/models/apply_job.rb:7]: khong thi thu xem di :smile:

File: app/models/apply_job.rb (hunk 2 above)
Comment (NgengChhengkim) [reply app/models/apply_job.rb:11]: ort yul mech ban ke ort oy brer find_by knong scope.  te find_by ta find khernh hoy leng find teat hoy. te ber where vinh find tol te os ban ghi

Comment (yongsokheng) [reply app/models/apply_job.rb:11]: where => select *
find_by => select \* limit 1
:smile:

**Pull request merged**